modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/operation/ImmutableProcessorChildContextChainExecutor.java

```diff
@@ -151,4 +155,18 @@ public List<Processor> getMessageProcessors() {
   public CoreEvent getOriginalEvent() {
     return originalEvent;
   }
+
+  private ComponentLocation calculateComponentLocation(ComponentLocation chainLocation) {
+    if (!(chainLocation instanceof DefaultComponentLocation)) {
+      return chainLocation;
+    }
+    // As per W-15158118, we need to remove the 'processors' added to the location of the scope, that identifies the chain
```

Comment:
why? Read that ticket and I don't find anything that justifies this. Or is it because of the tracing thing? really feels like you're hiding a bug here.

Reply:
The problem is that in this object we need the scope's location, which is an annotation. But adding the exact location to the component creates the re-registration of the component in the component location, when the chain is not the operation itself, but the processors within this. Unless we decide to create a different annotation for this, I don't see a way of having the location but not getting the other behavior

Reply:
(I am in fact proposing to create another annotation)